### PR TITLE
Fixed buffer overflow error when typed longer comment than at last qso

### PR DESCRIPTION
**Pull request by airween**
I ran into a problem in QSO mode when I typed a longer text in the comment field than 12 chars, and the call exists in list of previous contacts (aka `worked` list).

First, the available space in the WORKED window is 12 char wide, but the declaration contains only `char[12]`, so I incremented that value to 13.

Then I changed the `strcpy` by the `strncpy`, and set the max length. Finally, I set to `\0` at the max length position in the `worked[i].exchange`.

### CODE REVIEW

File: src/addcall.c [hunk 1]
```diff
@@ -93,7 +93,8 @@ int addcall(void) {
     j = getctydata(hiscall);
     worked[i].country = j;
     if (strlen(comment) >= 1) {		/* remember last exchange */
-	strcpy(worked[i].exchange, comment);
+	strncpy(worked[i].exchange, comment, 12);
```
Comment (zcsahok) [inline src/addcall.c:96]:
```
   g_strlcpy(worked[i].exchange, comment, sizeof(worked[i].exchange));
```
can replace both lines

**Review by zcsahok — COMMENTED** [review]
glibc's string functions are more safe

File: src/addcall.c (hunk 1 above)
Comment (airween) [reply src/addcall.c]: Thanks, fixed.

File: src/addcall.c [hunk 2]
```diff
@@ -93,8 +93,7 @@ int addcall(void) {
     j = getctydata(hiscall);
     worked[i].country = j;
     if (strlen(comment) >= 1) {		/* remember last exchange */
-	strncpy(worked[i].exchange, comment, 12);
-	worked[i].exchange[12] = 0;
+	g_strlcpy(worked[i].exchange, comment, 12);
```
Comment (dl1jbe) [inline src/addcall.c:96]: The last parameter should be 13 or better sizeof(worked[i].exchange). 

g_strlcpy needs the size of the buffer, not the number bytes you want to copy.

Comment (airween) [reply src/addcall.c]: TNX, fixed.

I always forget that glib functions can be used here :)

**Comment by dl1jbe** [comment]
Thanks. Applied and pushed.

**Pull request merged**